### ISSUE TITLE
Filter VMs step of wizard: revisit unintuitive VM count badges in tree view

### ISSUE THREAD

**Issue by mturley** [konveyor/forklift-ui#718]
See comment here: https://github.com/konveyor/forklift-ui/pull/675#discussion_r675110043

The numbers of VMs in the badges we show in the tree represent the count of ALL VMs from any selected descendants of that node, which means any direct VM children count towards the numbers on all their ancestors. This weird accumulation makes it seem like the math is wrong: you'll get 5 total VMs when you have two badges that say 5; at first glance you'd think there were 10 VMs, but it's just the same 5 VMs represented at two depths.

Possible solutions include better descriptions on the labels/badges, only including badges on leaf nodes (where the VMs actually come from -- possibly have an exception for collapsed nodes?), or maybe even getting rid of the badges except the one on the root node.

cc @vconzola let's chat about this some time. Not at all a priority.

**Comment by mturley** [comment]
Related note: in the folder tree, nodes that can contain direct VM children but are not presented as leaf nodes are selectable even if all their children are deselected. This is a weird and inconsistent experience, but it's because there can be uncategorized VMs that don't fit it any of their children. Maybe we can display an "uncategorized" folder/node or something similar so every selectable node is presented as a leaf.

**Comment by vconzola** [comment]
@mturley I thought we discussed this ad nauseum a while ago and came up with a solution we agreed to? What's changed? No need to discuss now. Let's talk about it when you're ready to work on this issue.

**Comment by mturley** [comment]
Yeah, we did. I just toured all this code with Spaxman and tried to explain it all and he raised the same questions :) maybe we can just call it done. I guess I didn't do a great job explaining the reasoning behind the decisions.

**Comment by seanforyou23** [comment]
Sorry bout that - just a new guy asking old questions 😅

**Comment by mturley** [comment]
There was never going to be a perfect solution, I think we just got tired of discussing it and landed on something that was good enough for 2.0 😆
